utils/streams.js

```diff
@@ -0,0 +1,171 @@
+  csvToJsonFile,
+  cssBundler
+};
+
+export default actions;
+
+if (require.main === module) {
+  cliRun();
+}
+
+function cliRun() {
+  const argv = Argv
+    .version('1.0.0')
+    .usage('Usage: $0 --action actionType [--file filePath]')
+    .example('$0 --action=io --file=./src/example.js')
+    .example('$0 --action transform --file=./data/test.txt')
+    .example('$0 -a csvToJsonStream -f ./data/test.csv')
+    .example('$0 -a csvToJsonFile -f ./data/test.csv')
+    .example('$0 --action cssBundler --path ./data/css')
+    .demandOption(['action'])
+    .option('action', {
+      alias: 'a',
+      describe: 'choose an action',
+      choices: ['io', 'transform', 'csvToJsonStream', 'csvToJsonFile', 'cssBundler']
+    })
+    .option('file', {
+      alias: 'f',
+      describe: 'provide a path to file (mandatory for -a=csvToJsonFile or -a=io)'
+    })
+    .option('path', {
+      alias: 'p',
+      describe: 'provide a path to folder to bundle CSS files (mandatory for --action=cssBundler)'
+    })
+    .help('h')
+    .argv;
+  
+  if (argv.action) {
+    if(argv.action === 'cssBundler') actions[argv.action](argv.path);
+    else actions[argv.action](argv.file);
+  }
+}
+
+/* stream helpers*/
+function inputOutput(filePath) { /* 4, babel-node ./utils/streams.js -a io -f ./data/test.txt */
+  if (fs.existsSync(filePath)) fs.createReadStream(filePath).pipe(process.stdout);
+  else console.log('Please specify correct file path');
+}
+
+function transform() { /* 5, node ./utils/streams.js -a transform */
+  stdInInformUser();
+  
+  let write = function (buffer, encoding, next) {
+    this.push(`Transformed to: ${buffer.toString().toUpperCase()}`);
+    next();
+  };
+  
+  process.stdin
+    .pipe(through(write, end))
+    .pipe(process.stdout);
+}
+
+function csvToJsonStream(filePath) { /* 6, babel-node ./utils/streams.js -a csvToJsonStream -f ./data/test.csv */
+  if (fs.existsSync(filePath) && path.extname(filePath) === '.csv') {
+    const readableStream = fs.createReadStream(filePath);
+    readableStream.setEncoding('utf8');
+    readableStream
+      .pipe(through(csvToJsonStreamTransform))
+      .pipe(process.stdout);
+  }
+  else console.log('Please specify correct file path');
+}
+
+function csvToJsonFile(filePath) { /* 7, babel-node ./utils/streams.js -a csvToJsonFile -f ./data/test.csv */
+  if (fs.existsSync(filePath) && path.extname(filePath) === '.csv') {
+    const readableStream = fs.createReadStream(filePath);
+    readableStream.setEncoding('utf8');
+    const fileName = createFilePath(filePath);
+    const writableStream = fs.createWriteStream(fileName);
+    readableStream
+      .pipe(through(csvToJsonStreamTransform, () => {
+        console.log(`${fileName} was generated`);
+      }))
+      .pipe(writableStream);
+  }
+  else console.log('Please specify correct file path');
+}
+
+function stdInInformUser() {
+  process.stdout.write(`Please start typing to transform ...
+  to end input use CTRL + D \n`);
```

Comment:
Ctrl+D doesn't work for me :(

Reply:
Please try Alt-Space. Then, C for Windows